From 7759cbf71ef73cfdab2700a82a92f2999a56d7c3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:25 -0400
Subject: [usb] xhci: Handle MaxPSASize == 0

Message-id: <1410887463-20674-314-git-send-email-dzickus@redhat.com>
Patchwork-id: 95585
O-Subject: [RHEL7 PATCH 313/591] xhci: Handle MaxPSASize == 0
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit f7920884eb640bc642f3b4e56f5237d30a080eda
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Nov 15 12:14:38 2013 +0100

    xhci: Handle MaxPSASize == 0

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 8d0bb87..2877c9a 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -3138,6 +3138,12 @@ int xhci_alloc_streams(struct usb_hcd *hcd, struct usb_device *udev,
  xhci_dbg(xhci, "Driver wants %u stream IDs (including stream 0).\n",
    num_streams);
 
+ /* MaxPSASize value 0 (2 streams) means streams are not supported */
+ if (HCC_MAX_PSA(xhci->hcc_params) < 4) {
+  xhci_dbg(xhci, "xHCI controller does not support streams.\n");
+  return -ENOSYS;
+ }
+
  config_cmd = xhci_alloc_command(xhci, true, true, mem_flags);
  if (!config_cmd) {
   xhci_dbg(xhci, "Could not allocate xHCI command structure.\n");
-- 
1.7.1